File: CS/ColumnBindingExample/ViewModels/MainViewModel.cs
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace ColumnBindingExample.ViewModels {
    [POCOViewModel]
    public class MainViewModel {
        protected MainViewModel() {
            Data = new ObservableCollection<Dictionary<string, object>>();
            for (int i = 0; i < 25; i++)
                Data.Add(new Dictionary<string, object>() { { "ID", i }, { "Name", "Name" + i }, { "Date", DateTime.Now.Date }, { "Check", false } });
            Columns = new ObservableCollection<ColumnViewModel>();
            Columns.Add(new ColumnViewModel() { Header = "ID", FieldName = "[ID]" });
            Columns.Add(new ColumnViewModel() { Header = "Name", FieldName = "[Name]" });
            Columns.Add(new ColumnViewModel() { Header = "Date", FieldName = "[Date]" });
            Columns.Add(new ColumnViewModel() { Header = "Check", FieldName = "[Check]" });
        }

        public virtual ObservableCollection<ColumnViewModel> Columns { get; set; }
        public virtual ObservableCollection<Dictionary<string, object>> Data { get; set; }
    }

    public class ColumnViewModel {
        public string FieldName { get; set; }
        public string Header { get; set; }
    }
}

File: VB/ColumnBindingExample/ViewModels/MainViewModel.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel

Namespace ColumnBindingExample.ViewModels
    <POCOViewModel> _
    Public Class MainViewModel
        Protected Sub New()
            Data = New ObservableCollection(Of Dictionary(Of String, Object))()
            For i As Integer = 0 To 24
                Data.Add(New Dictionary(Of String, Object)() From { _
                    { "ID", i }, _
                    { "Name", "Name" & i }, _
                    { "Date", Date.Now.Date }, _
                    { "Check", False } _
                })
            Next i
            Columns = New ObservableCollection(Of ColumnViewModel)()
            Columns.Add(New ColumnViewModel() With { _
                .Header = "ID", _
                .FieldName = "[ID]" _
            })
            Columns.Add(New ColumnViewModel() With { _
                .Header = "Name", _
                .FieldName = "[Name]" _
            })
            Columns.Add(New ColumnViewModel() With { _
                .Header = "Date", _
                .FieldName = "[Date]" _
            })
            Columns.Add(New ColumnViewModel() With { _
                .Header = "Check", _
                .FieldName = "[Check]" _
            })
        End Sub

        Public Overridable Property Columns() As ObservableCollection(Of ColumnViewModel)
        Public Overridable Property Data() As ObservableCollection(Of Dictionary(Of String, Object))
    End Class

    Public Class ColumnViewModel
        Public Property FieldName() As String
        Public Property Header() As String
    End Class
End Namespace